povray: add smoke test

ketsubouchi opened the pull request:
Add test function
Render `biscuit.pov` as `make check` and copy rendered `biscuit.png` to keep file.

Commit:
add test

Commit:
review

Commit:
import

Commit:
review

tldahlgren commented on var/spack/repos/builtin/packages/povray/package.py:
```diff
@@ -147,3 +147,9 @@ def configure_args(self):
+        povs = find(self.prefix.share, 'biscuit.pov')[0]
+        copy(povs, '.')
+        self.run_test('povray', options=['biscuit.pov'])
+        copy('biscuit.png', '..')
```
What is the purpose of this line?  How does it contribute to the smoke test?

tldahlgren commented on var/spack/repos/builtin/packages/povray/package.py:
```diff
@@ -147,3 +147,9 @@ def configure_args(self):
+    def test(self):
+        povs = find(self.prefix.share, 'biscuit.pov')[0]
+        copy(povs, '.')
+        self.run_test('povray', options=['biscuit.pov'])
```
It would be more helpful to add a `purpose` for the test so it gets output to the test log when the test is run.

tldahlgren commented on var/spack/repos/builtin/packages/povray/package.py:
```diff
@@ -147,3 +147,9 @@ def configure_args(self):
 
         return extra_args
+
+    def test(self):
```
Why not test the installation with the `--benchmark` option and check that the output matches a regular expression representation of at least part of the expected output?

tldahlgren replied:
Okay.  Good to know it isn't a sufficient test.

ketsubouchi commented:
Thank you for your kind suggestions.
1. What is the purpose of `copy('biscuit.png', '..')`
Sorry, I did not know `--keep-stage` option of `spack test run`
(We want result png file to compare with results of Windows binary.)
I removed this line.

2. add a purpose
I added a purpose.

3. Why not `--benchmark`
We want result file. `--benchmark` leaves no file and is not adequate.
I added `expected` normal end message.

Commit:
PR review

tldahlgren commented:
> Thank you for your kind suggestions.
> 
> 1. What is the purpose of `copy('biscuit.png', '..')`
>    Sorry, I did not know `--keep-stage` option of `spack test run`
>    (We want result png file to compare with results of Windows binary.)
>    I removed this line.

No problem.  That is a handy option.


> 2. add a purpose
>    I added a purpose.

Thanks!


> 3. Why not `--benchmark`
>    We want result file. `--benchmark` leaves no file and is not adequate.

Good to know.

>    I added `expected` normal end message.

Thanks!  I like to see output in the tests since it can be helpful for debugging.

tldahlgren commented:
Closing and re-opening the PR to force required tests to rerun.

tldahlgren reviewed (approved):
LGTM.  Confirmed new test passes for me locally.

Approval contingent on all required tests passing.
